pkg/synchronizer/kubernetes/cached_client.go

```diff
@@ -0,0 +1,36 @@
+package kubernetes
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type cachedClient struct {
+	clt    client.Client
+	cCache cache.Cache
```

Comment:
Sorry for being picky but I would just spell it out like `clientCache`.  I understand this but it will help any contributor who will maintain this code later.